### ISSUE TITLE
Missing boogieReservedNames

### ISSUE THREAD

**Issue by niomaster** [viperproject/carbon#418]
It seems that `BoogieNameGenerator.boogieReservedNames` is somewhat incomplete. I ran into this with e.g. `then` and `modifies`, but there are probably more:

```
$ cat ~/viper/carbonkeywords.vpr 
function then(): Int

$ ./bin/carbon ~/viper/carbonkeywords.vpr 
carbon 1.0
carbon found 2 errors in 1.64s:
  [0] An internal error occurred. Found an unparsable output from Boogie: /tmp/carbon14264761243206639989.bpl(181,11): error: ident expected (<no position>)
  [1] An internal error occurred. Found an unparsable output from Boogie: 1 parse errors detected in /tmp/carbon14264761243206639989.bpl (<no position>)
```

**Comment by gauravpartha** [comment]
Thanks for pointing this out. In https://github.com/viperproject/carbon/pull/425, I have added all keywords recorded in https://github.com/boogie-org/boogie/blob/3c9007eb5a933f6214ec84cbe5ccf6caf6dc8b4b/Source/Core/Scanner.cs#L514 that we had not captured before. Probably still some missing, but this should at least resolve this particular issue. Will close this issue once the PR has been merged.

**Comment by gauravpartha** [comment]
Closed by PR #425.
